[pkgconf] Build error on x64-linux

Package: pkgconf:x64-linux@2.4.3

**Host Environment**

- Host: x64-linux
- Compiler: GNU 10.2.1
-    vcpkg-tool version: 2025-04-16-f9b6c6917b23c1ccf16c1a9f015ebabf8f615045
    vcpkg-scripts version: e0b65b8cca 2025-04-20 (19 hours ago)

**To Reproduce**

`vcpkg install opencv ffmpeg boost-filesystem sqlite3 leveldb fmt drogon nlohmann-json --clean-after-build`

**Failure logs**

```
-- Found Python version '3.13.3 at /usr/local/bin/python3'
-- Using meson: /opt/vcpkg/downloads/tools/meson-1.7.2-549ef7/meson.py
Downloading https://github.com/pkgconf/pkgconf/archive/pkgconf-2.4.3.tar.gz -> pkgconf-pkgconf-pkgconf-2.4.3.tar.gz
Successfully downloaded pkgconf-pkgconf-pkgconf-2.4.3.tar.gz
-- Extracting source /opt/vcpkg/downloads/pkgconf-pkgconf-pkgconf-2.4.3.tar.gz
-- Using source at /opt/vcpkg/buildtrees/pkgconf/src/conf-2.4.3-0e28407cb4.clean
-- Found external ninja('1.12.1').
-- Configuring x64-linux-dbg
-- Getting CMake variables for x64-linux
-- Configuring x64-linux-dbg done
-- Configuring x64-linux-rel
-- Configuring x64-linux-rel done
-- Package x64-linux-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:127 (message):
    Command failed: /usr/local/bin/ninja install -v
    Working Directory: /opt/vcpkg/buildtrees/pkgconf/x64-linux-dbg
    Error code: 1
    See logs for more information:
      /opt/vcpkg/buildtrees/pkgconf/package-x64-linux-dbg-out.log

Call Stack (most recent call first):
  installed/x64-linux/share/vcpkg-tool-meson/vcpkg_install_meson.cmake:33 (vcpkg_execute_required_process)
  ports/pkgconf/portfile.cmake:67 (vcpkg_install_meson)
  scripts/ports.cmake:206 (include)



```

<details><summary>/opt/vcpkg/buildtrees/pkgconf/package-x64-linux-dbg-out.log</summary>

```
[1/21] /opt/rh/devtoolset-10/root/usr/bin/cc -fPIC -g -Ilibpkgconf.a.p -I. -I../src/conf-2.4.3-0e28407cb4.clean -I/opt/vcpkg/installed/x64-linux/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -g -D_BSD_SOURCE -D_DEFAULT_SOURCE -Wimplicit-function-declaration -Wmisleading-indentation -fPIC -g -fPIC -DLIBPKGCONF_EXPORT -DPKGCONFIG_IS_STATIC -MD -MQ libpkgconf.a.p/libpkgconf_audit.c.o -MF libpkgconf.a.p/libpkgconf_audit.c.o.d -o libpkgconf.a.p/libpkgconf_audit.c.o -c ../src/conf-2.4.3-0e28407cb4.clean/libpkgconf/audit.c
[2/21] /opt/rh/devtoolset-10/root/usr/bin/cc -fPIC -g -Ilibpkgconf.a.p -I. -I../src/conf-2.4.3-0e28407cb4.clean -I/opt/vcpkg/installed/x64-linux/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -g -D_BSD_SOURCE -D_DEFAULT_SOURCE -Wimplicit-function-declaration -Wmisleading-indentation -fPIC -g -fPIC -DLIBPKGCONF_EXPORT -DPKGCONFIG_IS_STATIC -MD -MQ libpkgconf.a.p/libpkgconf_fileio.c.o -MF libpkgconf.a.p/libpkgconf_fileio.c.o.d -o libpkgconf.a.p/libpkgconf_fileio.c.o -c ../src/conf-2.4.3-0e28407cb4.clean/libpkgconf/fileio.c
[3/21] /opt/rh/devtoolset-10/root/usr/bin/cc -fPIC -g -Ilibpkgconf.a.p -I. -I../src/conf-2.4.3-0e28407cb4.clean -I/opt/vcpkg/installed/x64-linux/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -g -D_BSD_SOURCE -D_DEFAULT_SOURCE -Wimplicit-function-declaration -Wmisleading-indentation -fPIC -g -fPIC -DLIBPKGCONF_EXPORT -DPKGCONFIG_IS_STATIC -MD -MQ libpkgconf.a.p/libpkgconf_bsdstubs.c.o -MF libpkgconf.a.p/libpkgconf_bsdstubs.c.o.d -o libpkgconf.a.p/libpkgconf_bsdstubs.c.o -c ../src/conf-2.4.3-0e28407cb4.clean/libpkgconf/bsdstubs.c
[4/21] /opt/rh/devtoolset-10/root/usr/bin/cc -fPIC -g -Ilibpkgconf.a.p -I. -I../src/conf-2.4.3-0e28407cb4.clean -I/opt/vcpkg/installed/x64-linux/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -g -D_BSD_SOURCE -D_DEFAULT_SOURCE -Wimplicit-function-declaration -Wmisleading-indentation -fPIC -g -fPIC -DLIBPKGCONF_EXPORT -DPKGCONFIG_IS_STATIC -MD -MQ libpkgconf.a.p/libpkgconf_buffer.c.o -MF libpkgconf.a.p/libpkgconf_buffer.c.o.d -o libpkgconf.a.p/libpkgconf_buffer.c.o -c ../src/conf-2.4.3-0e28407cb4.clean/libpkgconf/buffer.c
[5/21] /opt/rh/devtoolset-10/root/usr/bin/cc -fPIC -g -Ilibpkgconf.a.p -I. -I../src/conf-2.4.3-0e28407cb4.clean -I/opt/vcpkg/installed/x64-linux/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -g -D_BSD_SOURCE -D_DEFAULT_SOURCE -Wimplicit-function-declaration -Wmisleading-indentation -fPIC -g -fPIC -DLIBPKGCONF_EXPORT -DPKGCONFIG_IS_STATIC -MD -MQ libpkgconf.a.p/libpkgconf_argvsplit.c.o -MF libpkgconf.a.p/libpkgconf_argvsplit.c.o.d -o libpkgconf.a.p/libpkgconf_argvsplit.c.o -c ../src/conf-2.4.3-0e28407cb4.clean/libpkgconf/argvsplit.c
[6/21] /opt/rh/devtoolset-10/root/usr/bin/cc -fPIC -g -Ilibpkgconf.a.p -I. -I../src/conf-2.4.3-0e28407cb4.clean -I/opt/vcpkg/installed/x64-linux/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -g -D_BSD_SOURCE -D_DEFAULT_SOURCE -Wimplicit-function-declaration -Wmisleading-indentation -fPIC -g -fPIC -DLIBPKGCONF_EXPORT -DPKGCONFIG_IS_STATIC -MD -MQ libpkgconf.a.p/libpkgconf_fragment.c.o -MF libpkgconf.a.p/libpkgconf_fragment.c.o.d -o libpkgconf.a.p/libpkgconf_fragment.c.o -c ../src/conf-2.4.3-0e28407cb4.clean/libpkgconf/fragment.c
[7/21] /opt/rh/devtoolset-10/root/usr/bin/cc -fPIC -g -Ilibpkgconf.a.p -I. -I../src/conf-2.4.3-0e28407cb4.clean -I/opt/vcpkg/installed/x64-linux/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -g -D_BSD_SOURCE -D_DEFAULT_SOURCE -Wimplicit-function-declaration -Wmisleading-indentation -fPIC -g -fPIC -DLIBPKGCONF_EXPORT -DPKGCONFIG_IS_STATIC -MD -MQ libpkgconf.a.p/libpkgconf_client.c.o -MF libpkgconf.a.p/libpkgconf_client.c.o.d -o libpkgconf.a.p/libpkgconf_client.c.o -c ../src/conf-2.4.3-0e28407cb4.clean/libpkgconf/client.c
[8/21] /opt/rh/devtoolset-10/root/usr/bin/cc -fPIC -g -Ilibpkgconf.a.p -I. -I../src/conf-2.4.3-0e28407cb4.clean -I/opt/vcpkg/installed/x64-linux/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -g -D_BSD_SOURCE -D_DEFAULT_SOURCE -Wimplicit-function-declaration -Wmisleading-indentation -fPIC -g -fPIC -DLIBPKGCONF_EXPORT -DPKGCONFIG_IS_STATIC -MD -MQ libpkgconf.a.p/libpkgconf_personality.c.o -MF libpkgconf.a.p/libpkgconf_personality.c.o.d -o libpkgconf.a.p/libpkgconf_personality.c.o -c ../src/conf-2.4.3-0e28407cb4.clean/libpkgconf/personality.c
[9/21] /opt/rh/devtoolset-10/root/usr/bin/cc -fPIC -g -Ilibpkgconf.a.p -I. -I../src/conf-2.4.3-0e28407cb4.clean -I/opt/vcpkg/installed/x64-linux/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -g -D_BSD_SOURCE -D_DEFAULT_SOURCE -Wimplicit-function-declaration -Wmisleading-indentation -fPIC -g -fPIC -DLIBPKGCONF_EXPORT -DPKGCONFIG_IS_STATIC -MD -MQ libpkgconf.a.p/libpkgconf_cache.c.o -MF libpkgconf.a.p/libpkgconf_cache.c.o.d -o libpkgconf.a.p/libpkgconf_cache.c.o -c ../src/conf-2.4.3-0e28407cb4.clean/libpkgconf/cache.c
[10/21] /opt/rh/devtoolset-10/root/usr/bin/cc -fPIC -g -Ilibpkgconf.a.p -I. -I../src/conf-2.4.3-0e28407cb4.clean -I/opt/vcpkg/installed/x64-linux/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -g -D_BSD_SOURCE -D_DEFAULT_SOURCE -Wimplicit-function-declaration -Wmisleading-indentation -fPIC -g -fPIC -DLIBPKGCONF_EXPORT -DPKGCONFIG_IS_STATIC -MD -MQ libpkgconf.a.p/libpkgconf_parser.c.o -MF libpkgconf.a.p/libpkgconf_parser.c.o.d -o libpkgconf.a.p/libpkgconf_parser.c.o -c ../src/conf-2.4.3-0e28407cb4.clean/libpkgconf/parser.c
[11/21] /opt/rh/devtoolset-10/root/usr/bin/cc -fPIC -g -Ilibpkgconf.a.p -I. -I../src/conf-2.4.3-0e28407cb4.clean -I/opt/vcpkg/installed/x64-linux/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -g -D_BSD_SOURCE -D_DEFAULT_SOURCE -Wimplicit-function-declaration -Wmisleading-indentation -fPIC -g -fPIC -DLIBPKGCONF_EXPORT -DPKGCONFIG_IS_STATIC -MD -MQ libpkgconf.a.p/libpkgconf_path.c.o -MF libpkgconf.a.p/libpkgconf_path.c.o.d -o libpkgconf.a.p/libpkgconf_path.c.o -c ../src/conf-2.4.3-0e28407cb4.clean/libpkgconf/path.c
[12/21] /opt/rh/devtoolset-10/root/usr/bin/cc -fPIC -g -Ilibpkgconf.a.p -I. -I../src/conf-2.4.3-0e28407cb4.clean -I/opt/vcpkg/installed/x64-linux/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -g -D_BSD_SOURCE -D_DEFAULT_SOURCE -Wimplicit-function-declaration -Wmisleading-indentation -fPIC -g -fPIC -DLIBPKGCONF_EXPORT -DPKGCONFIG_IS_STATIC -MD -MQ libpkgconf.a.p/libpkgconf_dependency.c.o -MF libpkgconf.a.p/libpkgconf_dependency.c.o.d -o libpkgconf.a.p/libpkgconf_dependency.c.o -c ../src/conf-2.4.3-0e28407cb4.clean/libpkgconf/dependency.c
[13/21] /opt/rh/devtoolset-10/root/usr/bin/cc -fPIC -g -Ilibpkgconf.a.p -I. -I../src/conf-2.4.3-0e28407cb4.clean -I/opt/vcpkg/installed/x64-linux/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -g -D_BSD_SOURCE -D_DEFAULT_SOURCE -Wimplicit-function-declaration -Wmisleading-indentation -fPIC -g -fPIC -DLIBPKGCONF_EXPORT -DPKGCONFIG_IS_STATIC -MD -MQ libpkgconf.a.p/libpkgconf_pkg.c.o -MF libpkgconf.a.p/libpkgconf_pkg.c.o.d -o libpkgconf.a.p/libpkgconf_pkg.c.o -c ../src/conf-2.4.3-0e28407cb4.clean/libpkgconf/pkg.c
FAILED: libpkgconf.a.p/libpkgconf_pkg.c.o 
/opt/rh/devtoolset-10/root/usr/bin/cc -fPIC -g -Ilibpkgconf.a.p -I. -I../src/conf-2.4.3-0e28407cb4.clean -I/opt/vcpkg/installed/x64-linux/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -g -D_BSD_SOURCE -D_DEFAULT_SOURCE -Wimplicit-function-declaration -Wmisleading-indentation -fPIC -g -fPIC -DLIBPKGCONF_EXPORT -DPKGCONFIG_IS_STATIC -MD -MQ libpkgconf.a.p/libpkgconf_pkg.c.o -MF libpkgconf.a.p/libpkgconf_pkg.c.o.d -o libpkgconf.a.p/libpkgconf_pkg.c.o -c ../src/conf-2.4.3-0e28407cb4.clean/libpkgconf/pkg.c
../src/conf-2.4.3-0e28407cb4.clean/libpkgconf/pkg.c: In function ‘pkg_get_parent_dir’:
../src/conf-2.4.3-0e28407cb4.clean/libpkgconf/pkg.c:101:37: error: ‘O_DIRECTORY’ undeclared (first use in this function)
  101 |   const int dirfd = open(targetdir, O_DIRECTORY);
      |                                     ^~~~~~~~~~~
../src/conf-2.4.3-0e28407cb4.clean/libpkgconf/pkg.c:101:37: note: each undeclared identifier is reported only once for each function it appears in
../src/conf-2.4.3-0e28407cb4.clean/libpkgconf/pkg.c:106:17: warning: implicit declaration of function ‘readlinkat’; did you mean ‘readlink’? [-Wimplicit-function-declaration]
  106 |   ssize_t len = readlinkat(dirfd, targetfilename, sourcebuf, sizeof(sourcebuf) - 1);
      |                 ^~~~~~~~~~
      |                 readlink
[14/21] /opt/rh/devtoolset-10/root/usr/bin/cc -fPIC -g -Ilibpkgconf.a.p -I. -I../src/conf-2.4.3-0e28407cb4.clean -I/opt/vcpkg/installed/x64-linux/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -g -D_BSD_SOURCE -D_DEFAULT_SOURCE -Wimplicit-function-declaration -Wmisleading-indentation -fPIC -g -fPIC -DLIBPKGCONF_EXPORT -DPKGCONFIG_IS_STATIC -MD -MQ libpkgconf.a.p/libpkgconf_tuple.c.o -MF libpkgconf.a.p/libpkgconf_tuple.c.o.d -o libpkgconf.a.p/libpkgconf_tuple.c.o -c ../src/conf-2.4.3-0e28407cb4.clean/libpkgconf/tuple.c
[15/21] /opt/rh/devtoolset-10/root/usr/bin/cc -fPIC -g -Ilibpkgconf.a.p -I. -I../src/conf-2.4.3-0e28407cb4.clean -I/opt/vcpkg/installed/x64-linux/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -g -D_BSD_SOURCE -D_DEFAULT_SOURCE -Wimplicit-function-declaration -Wmisleading-indentation -fPIC -g -fPIC -DLIBPKGCONF_EXPORT -DPKGCONFIG_IS_STATIC -MD -MQ libpkgconf.a.p/libpkgconf_queue.c.o -MF libpkgconf.a.p/libpkgconf_queue.c.o.d -o libpkgconf.a.p/libpkgconf_queue.c.o -c ../src/conf-2.4.3-0e28407cb4.clean/libpkgconf/queue.c
[16/21] /opt/rh/devtoolset-10/root/usr/bin/cc -fPIC -g -Ipkgconf.p -I. -I../src/conf-2.4.3-0e28407cb4.clean -I/opt/vcpkg/installed/x64-linux/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -g -D_BSD_SOURCE -D_DEFAULT_SOURCE -Wimplicit-function-declaration -Wmisleading-indentation -fPIC -g -DPKGCONFIG_IS_STATIC -MD -MQ pkgconf.p/cli_renderer-msvc.c.o -MF pkgconf.p/cli_renderer-msvc.c.o.d -o pkgconf.p/cli_renderer-msvc.c.o -c ../src/conf-2.4.3-0e28407cb4.clean/cli/renderer-msvc.c
[17/21] /opt/rh/devtoolset-10/root/usr/bin/cc -fPIC -g -Ipkgconf.p -I. -I../src/conf-2.4.3-0e28407cb4.clean -I/opt/vcpkg/installed/x64-linux/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -g -D_BSD_SOURCE -D_DEFAULT_SOURCE -Wimplicit-function-declaration -Wmisleading-indentation -fPIC -g -DPKGCONFIG_IS_STATIC -MD -MQ pkgconf.p/cli_getopt_long.c.o -MF pkgconf.p/cli_getopt_long.c.o.d -o pkgconf.p/cli_getopt_long.c.o -c ../src/conf-2.4.3-0e28407cb4.clean/cli/getopt_long.c
[18/21] /opt/rh/devtoolset-10/root/usr/bin/cc -fPIC -g -Ipkgconf.p -I. -I../src/conf-2.4.3-0e28407cb4.clean -I/opt/vcpkg/installed/x64-linux/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -g -D_BSD_SOURCE -D_DEFAULT_SOURCE -Wimplicit-function-declaration -Wmisleading-indentation -fPIC -g -DPKGCONFIG_IS_STATIC -MD -MQ pkgconf.p/cli_main.c.o -MF pkgconf.p/cli_main.c.o.d -o pkgconf.p/cli_main.c.o -c ../src/conf-2.4.3-0e28407cb4.clean/cli/main.c
ninja: build stopped: subcommand failed.
```
</details>

--- a/ports/pkgconf/portfile.cmake
+++ b/ports/pkgconf/portfile.cmake
@@ -1,9 +1,18 @@
+vcpkg_download_distfile(
+    ADD_MISSING_DEFINE_PATCH
+    URLS https://github.com/pkgconf/pkgconf/commit/664b53d5c4920e66e4a57c7515ccfd1bd1477bac.patch?full_index=1
+    FILENAME pkgconf-add-missing-define-664b53d5c4920e66e4a57c7515ccfd1bd1477bac.patch
+    SHA512 1e5dc8b6ac9547157694c1674660c0f207876ed2916a053217c9299eb352c05d1f03a1ba8afddaf994b7c22474590407c627846cd68000bf53bd7229d9009895
+)
+
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO pkgconf/pkgconf
     REF "pkgconf-${VERSION}"
     SHA512 8c73b8f9c3dd3c72e6b0acf139a68054d631484af4618f46c30c3c13947294986d809bbac0648af9e974739fd42e1730a4e22323884d7cf72f0843a972991a99
     HEAD_REF master
+    PATCHES
+        "${ADD_MISSING_DEFINE_PATCH}"
 )
 
 vcpkg_configure_meson(

--- a/ports/pkgconf/vcpkg.json
+++ b/ports/pkgconf/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "pkgconf",
   "version": "2.4.3",
+  "port-version": 1,
   "description": "pkgconf is a program which helps to configure compiler and linker flags for development libraries. It is similar to pkg-config from freedesktop.org.",
   "homepage": "https://github.com/pkgconf/pkgconf",
   "license": null,

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -7222,7 +7222,7 @@
     },
     "pkgconf": {
       "baseline": "2.4.3",
-      "port-version": 0
+      "port-version": 1
     },
     "plasma-wayland-protocols": {
       "baseline": "1.14.0",

--- a/versions/p-/pkgconf.json
+++ b/versions/p-/pkgconf.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "ee86667e248e39cfb0ac994ac4e0bd00ce5b0836",
+      "version": "2.4.3",
+      "port-version": 1
+    },
+    {
       "git-tree": "a9135b2eb2d40c4f0fb2884160a9c6c8a3cd492d",
       "version": "2.4.3",
       "port-version": 0

